lib/mparticle_flutter_sdk_web.dart

```diff
@@ -260,6 +261,119 @@ class MparticleFlutterSdkWeb {
+
+        var transactionAttributes = {};
+        if (commerceEvent['transactionAttributes'] != null) {
+          transactionAttributes = commerceEvent['transactionAttributes'];
+        }
+
+        String? checkoutStep = commerceEvent['checkoutStep'];
+        if (checkoutStep != null) {
+          transactionAttributes['Step'] = checkoutStep;
+        }
+
+        String? checkoutOptions = commerceEvent['checkoutOptions'];
+        if (checkoutOptions != null) {
+          transactionAttributes['Option'] = checkoutOptions;
+        }
+
+        List? rawProducts = commerceEvent['products'];
+        List? products = [];
+        if (rawProducts != null && rawProducts.length > 0) {
+          rawProducts.forEach((rawProduct) {
+            var product = mpCommerce.callMethod('createProduct', [
+              rawProduct['name'],
+              rawProduct['quantity'],
+              rawProduct['price']
+            ]);
+            products.add(product);
+          });
+        }
+
+        int? productActionType = commerceEvent['jsProductActionType'];
+        int? promotionActionType = commerceEvent['jsPromotionActionType'];
+
+        // log product action
+        if (productActionType != null) {
+          mpCommerce.callMethod('logProductAction', [
+            productActionType,
+            JsObject.jsify(products),
+            JsObject.jsify(customAttributes),
+            JsObject.jsify(customFlags),
+            JsObject.jsify(transactionAttributes)
+          ]);
+          return true;
+          // log promotion
+        } else if (promotionActionType != null) {
+          List? rawPromotions = commerceEvent["promotions"];
+          List? promotions = [];
+
+          if (rawPromotions != null && rawPromotions.length > 0) {
+            rawPromotions.forEach((rawPromotion) {
+              var promotion = mpCommerce.callMethod('createPromotion', [
+                rawPromotion['promotionId'],
+                rawPromotion['creative'],
+                rawPromotion['name'],
+                rawPromotion['position'],
+              ]);
+              promotions.add(promotion);
+            });
+          }
+
+          mpCommerce.callMethod('logPromotion',
+              [promotionActionType, JsObject.jsify(promotions)]);
+
+          return true;
+          // log impression
+        } else {
+          List? rawImpressions = commerceEvent["impressions"];
+          List? impressions = [];
+
+          if (rawImpressions != null && rawImpressions.length > 0) {
+            rawImpressions.forEach((rawImpression) {
+              List impressionProducts = [];
+              var rawImpressionProducts = rawImpression['products'];
+
+              if (rawImpressionProducts != null &&
+                  rawImpressionProducts.length > 0) {
+                rawImpressionProducts.forEach((rawImpressionProduct) {
+                  var product = mpCommerce.callMethod('createProduct', [
+                    rawImpressionProduct['name'],
```

Comment:
is this missing "sku"? same for the `creatProduct` call on line 303

Reply:
Good catch.  The order of arguments is also incorrect, I'll update.